### PR TITLE
Skip setting XAML layout when tree is collapsed

### PR DESCRIPTION
**Pull request by rozele**
### Type of Change
_Erase all that don't apply._
- Performance optimization

### Why
When a view tree is collapsed (via `display: 'none'` style setting), there is no need to update the ViewPanel positioning properties (Canvas::TopProperty, Canvas::LeftProperty, FrameworkElement::WidthProperty, and FrameworkElement::HeightProperty), which would in turn trigger invalidate XAML layout for ViewPanels in that tree.

### What
This change ensures we do not update these XAML positioning properties when a view is in the context of a collapsed sub-tree.

## Testing
Ran `display: none` API test in RNTester:

https://user-images.githubusercontent.com/1106239/194947794-bac1a4b9-476c-4b2f-b623-2d475817733e.mp4

_Optional_: Describe the tests that you ran locally to verify your changes.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10712)

### CODE REVIEW

**Comment by rozele** [comment]
One additional minor change - we'll need to set all the measurement results based on Yoga values (rather than XAML ActualWidth values).

File: vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp [hunk 1]
```diff
@@ -1011,15 +1007,21 @@ void NativeUIManager::measure(
     }
   }
 
-  winrt::Rect rectInParentCoords = GetRectOfElementInParentCoords(feView, feRootView);
+  auto x = 0.f;
+  auto y = 0.f;
```
Comment (rozele) [inline vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp:1011]: Can simplify to `winrt::Point position = {0, 0};`

**Review by chiaramooney — COMMENTED** [review]
Looks good to me. Tagging @chrisglein to take a look for final eval on layout code from XAML perspective.

File: vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp [hunk 2]
```diff
@@ -928,9 +932,10 @@ void NativeUIManager::SetLayoutPropsRecursive(int64_t tag) {
     float top = YGNodeLayoutGetTop(yogaNode);
     float width = YGNodeLayoutGetWidth(yogaNode);
     float height = YGNodeLayoutGetHeight(yogaNode);
-    auto view = shadowNode.GetView();
-    auto pViewManager = shadowNode.GetViewManager();
-    pViewManager->SetLayoutProps(shadowNode, view, left, top, width, height);
+    const auto uiElement = view.try_as<xaml::UIElement>();
+    if (!isCollapsed) {
+      pViewManager->SetLayoutProps(shadowNode, view, left, top, width, height);
+    }
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp:938]: Seems like this should have a comment as to why we don't update the layout props. As in this is not a matter of correctness but is a performance optimization to reduce layout invalidations.

**Comment by chrisglein** [comment]
> * Performance optimization

What kind of measured impact have you seen from the change? Which scenarios was this causing the most unnecessary churn?

> One additional minor change - we'll need to set all the measurement results based on Yoga values (rather than XAML ActualWidth values).

Is there any potential side effect to no longer using the XAML layout results? Will this hide potential drift between Yoga values and the results applied to the XAML composition tree?

> This change ensures we do not update these XAML positioning properties when a view is in the context of a collapsed sub-tree.

Do we have full confidence that when things are un-collapsed that everything is appropriately being invalidated and pushed through again?

**Comment by chiaramooney** [comment]
Discussed offline with the team, and at this time we are going to opt not to take this change. Since the change does not offer a clear performance benefit, we are going to favor avoiding risk of layout breaks to our customers. Closing the PR at this time.

**Comment by rozele** [comment]
Thanks @chiaramooney. It's probably a good call. This will be much harder to accomplish in Fabric anyway :). We really wouldn't need this change if there wasn't the cached flex basis bug in Yoga, and it's primary purpose is to support a native module we use in Messenger to detect when text gets trimmed and force re-measure, which is not something available to open source / core React Native anyway.

The riskier bits of this change (namely, changes to `measure` APIs) are closer to how Fabric will work (using Yoga layout results rather than XAML results). In terms of performance benefit, we are seeing some early positive results for a similar change we made where we avoid setting Top and Left layout values if they have not changed (to avoid calling `InvalidateArrange()` in ViewPanel), so there may be performance benefits to this for apps that use Suspense quite frequently (where they'd run into `display: 'none'` more often).

**Comment by rozele** [comment]
@chrisglein - a few changes landed in iOS and Android to restore the ScrollView offset before and after setting `display: 'none'` on a ScrollView. The problem was that, when you set `display: 'none'`, Yoga will force the view to 0w x 0h dimensions, and the scroll offset will get reset to 0. When the view returns, it's offset will stay at 0.

The same can be said about XAML, and we should probably consider a similar fix at some point. Just dropping in here to point out that *not* setting the Yoga dimensions on display: 'none' sub-trees would also fix this issue.

There was also some appetite to apply a similar "fix" to Fabric to retain the scroll offset position, but I think it was deemed easier just to reset the scroll position after transitioning between display: 'none' and display: 'flex'

**Pull request closed without merging**